src/main/java/org/opensearch/neuralsearch/query/HybridQueryBuilder.java

```diff
@@ -254,6 +282,9 @@ protected boolean doEquals(HybridQueryBuilder obj) {
         }
         EqualsBuilder equalsBuilder = new EqualsBuilder();
         equalsBuilder.append(queries, obj.queries);
+        if (isClusterOnOrAfterMinReqVersionForPaginationInHybridQuery()) {
```

Comment:
in what scenario we need to do this check, and what will happen if code does not have it?

Reply:
I put it to maintain the consistency in the code.

Reply:
Because this field is introduced in version 2.19, for code understanding purpose and keeping the consistency, I put this check

Reply:
then why we need it in hashCode method?